nexus/src/db/queries/external_ip.rs

```diff
@@ -392,23 +393,34 @@ impl NextExternalIp {
     // each IP Pool, along with the pool/range IDs.
     //
     // ```sql
-    // SELECT
-    //     ip_pool_id,
-    //     id AS ip_pool_range_id,
-    //     first_address +
-    //         generate_series(0, last_address - first_address)
-    //         AS candidate_ip
-    // FROM
-    //     ip_pool_range
-    // WHERE
-    //     <pool_restriction> AND
-    //     time_deleted IS NULL
+    // SELECT * FROM (
+    //   SELECT
```

Comment:
If you're curious about how this works, there's an example fiddle which shows the logic:

https://www.db-fiddle.com/f/4jyoMCicNSZpjMt4jFYoz5/7203

Reply:
So I admittedly struggled a bit with this query, because:

- On the one hand, I really want to have a "unified allocation flow" for IP allocation. I don't want to, for example, forget to update the collection rcgen for service IPs vs instance IPs...
- ... but on the other hand, yeah, scanning through addresses when we know we want to allocate *one* is kinda non-optimal.

I went ahead and used this optimized form in 1637370b20978dab16b5bc0f6efa08d3b6c9a0a4 , which more-or-less matches the fiddle you provided.